crates/policy-evaluator/src/policy_evaluator.rs

```diff
@@ -14,31 +14,42 @@ use chimera_kube_policy_sdk::settings::SettingsValidationResponse;
 use crate::utils::convert_yaml_map_to_json;
 use crate::validation_response::ValidationResponse;
 
+use crate::cluster_context::ClusterContext;
+
+pub(crate) fn host_callback(
+    _id: u64,
+    binding: &str,
+    namespace: &str,
+    _operation: &str,
+    _payload: &[u8],
+) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>> {
+    let cluster_context = ClusterContext::get();
+    if binding != "kubernetes" {
+        return Err(format!("unknown binding: {}", binding).into());
+    }
+    match namespace {
+        "ingresses" => Ok(cluster_context.ingresses().into()),
+        "namespaces" => Ok(cluster_context.namespaces().into()),
+        "services" => Ok(cluster_context.services().into()),
+        _ => Err(format!("unknown namespace name: {}", namespace).into()),
+    }
+}
+
 pub struct PolicyEvaluator {
     wapc_host: WapcHost,
     settings: serde_json::Map<String, serde_json::Value>,
 }
 
 impl PolicyEvaluator {
-    pub fn new(
-        wasm_file: String,
-        settings: serde_yaml::Mapping,
-        host_callback: impl Fn(
-                u64,
-                &str,
-                &str,
-                &str,
-                &[u8],
-            )
-                -> std::result::Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>>
-            + 'static
-            + Sync
-            + Send,
-    ) -> Result<PolicyEvaluator> {
+    pub fn new(wasm_file: String, settings: serde_yaml::Mapping) -> Result<PolicyEvaluator> {
         let mut f = File::open(&wasm_file)?;
         let mut buf = Vec::new();
         f.read_to_end(&mut buf)?;
 
+        if let Err(error) = ClusterContext::init() {
+            println!("non fatal error: could not initialize a cluster context due to error: {}; context sensitive functions will not return any information", error);
+        }
```

Comment:
I don't think this will ever happen because the `init` code `panic!`(s) when an error happens.

Reply:
Yes, this is a leftover and I'm still divided on how to best handle this case. It's tricky because something could go wrong when initializing (e.g. could not find any usable configuration to build a Kube client -- no endless loop) vs "everything went fine" meaning we are in an endless loop polling for certain resources. I think I'll just remove this if and leave the inner panic/fatal_error.

Reply:
It's also worth mentioning that a failure to create a Kube client should not be fatal and the policy-server/testdrive can *still* serve or load policies. The only issue is that if they try to read contextual resources they will be empty, but if they are "offline" checks they will work perfectly.

Reply:
I'm all for exiting with an error if it's not possible to create a k8s client. If possible, I would however prefer to raise an error and then handle the exit inside of the `main`.

Reply:
If we exit with an error then there's no possibility to run offline policies, do we want that? Also, I don't think there's a way to exit inside of the main -- we'd have to join the thread, but if everything went fine, the thread we are joining is an endless loop... Am I missing anything?

Reply:
I think we have to work on defining a story around this type of failure, I have too many questions:
  * Should policy-server exit immediately with a failure when the client can't be created?
  * Should instead policy-server keep running, print a warning message and just serve empty results about `Namespace`, `Service`,... whatever?
  * Should policy-server look at the policies it has to host and act differently if none of the policies it loaded are "context aware" ones?

:exploding_head: too many variables... I can go on with other possibilities.

I would say, let's keep things simple and just panic